.github/workflows/test-package.yml

```diff
@@ -0,0 +1,35 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Test package
+
+on:
+  [workflow_dispatch]
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.4','3.5','3.6','3.7', '3.8', '3.9', '3.10']
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install tox tox-gh-actions pre-commit
+    - name: Run pre-commit on all files
+      run: |
+        pre-commit run --all-files
+    - name: Test with tox
+      env:
+        FEDERATION_URL: ${{ secrets.FEDERATION_URL }}
+        API_KEY: ${{ secrets.API_KEY }}
+        MM_API_KEY: ${{ secrets.MM_API_KEY }}
+        FABRIC: ${{ secrets.FABRIC }}
+      run: tox
```

Comment:
We can add url_tester too to test whether all links used in .md files are active or not this will inform us about broken links whenever a PR is made and any PR which has broken links in md files.. It will give a RuntimeError("Above pages have broken links") if links are broken

Reply:
I thought that we will run it with all the tests or that is not the case? Yea, I obviously need to add new links as well

Reply:
with pytest commant url_tester won't work.. we need to run python3 tests/url_tester.py to run it from root folder

Reply:
should we add it as part of pytest or should we keep it as a separate script to test urls (I made a separate script since it was not connected to our source codes)?

Reply:
I think you did a great job separating that script since we are not testing the core platform. I will add it to run in the pipeline. Let me check that and we can talk about how to add it to the pipeline. @koshy-thomas